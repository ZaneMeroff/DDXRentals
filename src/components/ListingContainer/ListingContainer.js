import React from 'react';
import './ListingContainer.scss'
import SmallListingCard from '../SmallListingCard/SmallListingCard.js';
import tubes from '../../data/menu-tubes.png'

const ListingContainer = (props) => {
  const allListings = props.listings.map(place => {
    return <SmallListingCard
      id={place.listing_id}
      area={place.area_id}
      name={place.name}
      address={place.address.street}
      addressZip={place.address.zip}
      cost={place.details.cost_per_night}
      features={place.details.features}
      beds={place.details.beds}
      baths={place.details.baths}
      addFavorite={props.addFavorite}
    />
  })
  return (
    <section className='listing-container'>
      {allListings}
      <img id='tubes' src={ tubes } alt='tubular menu for scrolling' />
    </section>
  )
}

export default ListingContainer;
