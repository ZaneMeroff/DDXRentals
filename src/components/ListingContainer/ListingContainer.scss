.listing-container {
  background-color: #202020;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 80px;
}

#tubes {
  position: fixed;
  height: 80%;
  left: 85%;
  z-index: 10;
  width: 150px;
  object-fit: cover;
}
