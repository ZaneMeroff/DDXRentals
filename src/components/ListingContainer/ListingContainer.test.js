import React from 'react';
import ReactDOM from 'react-dom';
import ListingContainer from './ListingContainer';
import { shallow } from 'enzyme';

describe('ListingContainer', () => {
  let wrapper, mockAddFavorite, mockArea;

  beforeEach(() => {
    mockArea =  [{
      listing_id: 3, area_id: 590, name: 'Hip RiNo Party Spot',
      address: {
        street: '2250 Lawrence St',
        zip: '80205'
      },
      details: {
        neighborhood_id: 5124122, superhost: true, seller_source: '91jss1', beds: 3, baths: 2.5, cost_per_night: 420, features: [
          'hot tub',
          'espresso machine'
        ],
      },
      dev_id: 'u4gh2j', area: 'rino', db_connect: 834470
    }]
  })

  it('should be an instance of ListingContainer component', () => {
    mockAddFavorite = jest.fn()
    wrapper = shallow(<ListingContainer
        addFavorite={mockAddFavorite}
        listings={mockArea}
      />);
      expect(wrapper).toMatchSnapshot();
  });

});
