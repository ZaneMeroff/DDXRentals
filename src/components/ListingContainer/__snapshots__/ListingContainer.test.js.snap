// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingContainer should be an instance of ListingContainer component 1`] = `
<section
  className="listing-container"
>
  <SmallListingCard
    addFavorite={[MockFunction]}
    address="2250 Lawrence St"
    addressZip="80205"
    area={590}
    baths={2.5}
    beds={3}
    cost={420}
    features={
      Array [
        "hot tub",
        "espresso machine",
      ]
    }
    id={3}
    name="Hip RiNo Party Spot"
  />
  <img
    alt="tubular menu for scrolling"
    id="tubes"
    src="menu-tubes.png"
  />
</section>
`;
